data/src/main/java/com/android254/data/work/SyncDataWorker.kt

```diff
@@ -58,87 +53,35 @@ class SyncDataWorker @AssistedInject constructor(
         withContext(ioDispatcher) {
-            val sessionSyncDefferred = async { syncSessions() }
-            val speakersDeffered = async { syncSpeakers() }
-            val sponsorsDeffered = async { syncSponsors() }
-
-            sessionSyncDefferred.await()
-            speakersDeffered.await()
-            sponsorsDeffered.await()
+            awaitAll(
+                async { sessionsRepo.syncSessions() },
+                async { speakersRepo.syncSpeakers() },
+                async { sponsorsRepo.syncSponsors() },
+                async { organizersRepo.syncOrganizers() }
+            )
         }
-
         return Result.success()
     }
 
-    private suspend fun syncSessions() {
-        withContext(ioDispatcher) {
-            val response = remoteSessionsDataSource.getAllSessionsRemote()
-            when (response) {
-                is ResourceResult.Success -> {
-                    localSessionsDataSource.deleteCachedSessions()
-                    localSessionsDataSource.saveCachedSessions(
-                        sessions = response.data ?: emptyList()
-                    )
-                    Timber.d("Sync sessions successful")
-                }
-
-                is ResourceResult.Error -> {
-                    Timber.d("Sync sessions failed ${response.message}")
-                }
-
-                else -> {}
-            }
-        }
-    }
-
-    private suspend fun syncSponsors() {
-        withContext(ioDispatcher) {
-            val response = remoteSponsorsDataSource.getAllSponsorsRemote()
-            when (response) {
-                is ResourceResult.Success -> {
-                    localSponsorsDataSource.deleteCachedSponsors()
-                    localSponsorsDataSource.saveCachedSponsors(
-                        sponsors = response.data ?: emptyList()
-                    )
-                    Timber.d("Sync sponsors successful")
-                }
-
-                is ResourceResult.Error -> {
-                    Timber.d("Sync sponsors failed ${response.message}")
-                }
-
-                else -> {}
-            }
-        }
-    }
-
-    private suspend fun syncSpeakers() {
-        withContext(ioDispatcher) {
-            val response = remoteSpeakersDataSource.getAllSpeakersRemote()
-            when (response) {
-                is ResourceResult.Success -> {
-                    localSpeakersDataSource.deleteAllCachedSpeakers()
-                    localSpeakersDataSource.saveCachedSpeakers(
-                        speakers = response.data ?: emptyList()
-                    )
-
-                    Timber.d("Sync speakers successful")
-                }
-
-                is ResourceResult.Error -> {
-                    Timber.d("Sync speakers failed ${response.message}")
-                }
-
-                else -> {}
-            }
-        }
-    }
+//    private suspend fun startForegroundService(notificationInfo:String) {
```

Comment:
Maybe cleanup if this is no longer needed?

Reply:
Let me remove it 
I forgot